viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/NodesAndJunctionsService.scala

```diff
@@ -1186,8 +1186,26 @@ class NodesAndJunctionsService(roadwayDAO: RoadwayDAO, roadwayPointDAO: RoadwayP
               nodePointDAO.insertCalculatedNodePoint(rwPoint, BeforeAfter.After, nodeNumber, username)
               nodePointCount = nodePointCount + 2
             } else {
-              nodePointDAO.insertCalculatedNodePoint(rwPoint, beforeAfterValue, nodeNumber, username)
-              nodePointCount = nodePointCount + 1
+              val nodePointTemplates = nodePointDAO.fetchNodePointTemplateByRoadwayPointId(rwPoint).headOption
+              if (nodePointTemplates.nonEmpty) {
+                val nodePointTemplate = nodePointTemplates.head
+
+                if (nodePointTemplate.nodePointType == NodePointType.CalculatedNodePoint) {
```

Comment:
Tämän if-elsen sanallinen yleistason kommentointi helpottaisi ymmärtämistä.
Nyt jää ilmaan, miksi näin (vaikka koodista toki näkee, että toisessa tehdään CalculatedNP ja toisessa RoadNP - mutta miksi)